### DISCUSSION TITLE
neurokit method of reference in ecg_clean

### DISCUSSION THREAD

**Discussion by alexxony** [neuropsychology/NeuroKit#548] [Q&A] [answered]
**Question and context**
I would like to find the reference paper of  neurokit method of reference.

Do you apply any existing theory on your method?

 

>   def _ecg_clean_nk(ecg_signal, sampling_rate=1000):
>   
>       # Remove slow drift and dc offset with highpass Butterworth.
>       clean = signal_filter(signal=ecg_signal, sampling_rate=sampling_rate, lowcut=0.5, method="butterworth", order=5)
>   
>       clean = signal_filter(signal=clean, sampling_rate=sampling_rate, method="powerline", powerline=50)
>       return clean
>

**Comment by DominiqueMakowski** [comment, marked as answer]
You can reference the package generally as: 

```You can cite NeuroKit2 as follows:

- Makowski, D., Pham, T., Lau, Z. J., Brammer, J. C., Lespinasse, F., Pham, H.,
Schölzel, C., & Chen, S. A. (2021). NeuroKit2: A Python toolbox for neurophysiological signal processing.
Behavior Research Methods, 53(4), 1689–1696. https://doi.org/10.3758/s13428-020-01516-y

Full bibtex reference:

@article{Makowski2021neurokit,
    author = {Dominique Makowski and Tam Pham and Zen J. Lau and Jan C. Brammer and Fran{\c{c}}ois Lespinasse and Hung Pham and Christopher Schölzel and S. H. Annabel Chen},
    title = {{NeuroKit}2: A Python toolbox for neurophysiological signal processing},
    journal = {Behavior Research Methods},
    volume = {53},
    number = {4},
    pages = {1689--1696},
    publisher = {Springer Science and Business Media {LLC}},
    doi = {10.3758/s13428-020-01516-y},
    url = {https://doi.org/10.3758%2Fs13428-020-01516-y},
    year = 2021,
    month = {feb}
}
```

We don't have specific references for these cleaning steps
